truncate_client/src/lil_bits/daily_splash/graph.rs

```diff
@@ -0,0 +1,183 @@
+        let moves: u32 = today.attempts.iter().map(|a| a.moves).sum();
+
+        let today_label = format!(
+            "{}: {} attempt{}, {} total move{}",
+            if won { "Won" } else { "Lost" },
+            attempts,
+            if attempts == 1 { "" } else { "s" },
+            moves,
+            if moves == 1 { "" } else { "s" },
+        );
+        let today_height: f32 = moves as f32 / max_total_moves as f32;
+
+        Self {
+            moves_graph_texture,
+            streak_graph_texture,
+            date_labels: (leftmost_date_string, rightmost_date_string),
+            highlighted_day: Some(DayHighlight {
+                day_from_right: 0,
+                label: today_label,
+                bar_height: today_height,
+            }),
+            days_played,
+        }
+    }
+
+    pub fn render(&self, ui: &mut egui::Ui) {
+        let fz = 16.0; // Label font size, should come from a theme.
+        let (graph_rect, _) =
+            ui.allocate_exact_size(vec2(ui.available_width(), 128.0), Sense::hover());
+
+        let mut moves_graph_rect = graph_rect.clone();
+        *moves_graph_rect.top_mut() += 30.0;
+        *moves_graph_rect.bottom_mut() -= 18.0;
+
+        let mut streak_graph_rect = moves_graph_rect.clone();
+        streak_graph_rect.set_bottom(moves_graph_rect.bottom() + 6.0);
+        streak_graph_rect.set_top(moves_graph_rect.bottom() + 2.0);
+
+        let date = TextHelper::light(&self.date_labels.1, fz, None, ui);
+        date.paint_within(graph_rect, Align2::RIGHT_BOTTOM, Color32::WHITE, ui);
+
+        let date = TextHelper::light(&self.date_labels.0, fz, None, ui);
+        date.paint_within(graph_rect, Align2::LEFT_BOTTOM, Color32::WHITE, ui);
+
+        if let Some(highlight) = &self.highlighted_day {
+            let notes = TextHelper::light(&highlight.label, fz, None, ui);
+            // TODO: This will need to align to the correct day when highlighting !today
+            notes.paint_within(graph_rect, Align2::RIGHT_TOP, Color32::WHITE, ui);
+
+            let day_column_width = graph_rect.width() / self.days_played as f32;
+            let right_offset =
+                (highlight.day_from_right as f32) * day_column_width + (day_column_width / 2.0);
+
+            let line_start = graph_rect.right_top() + vec2(-right_offset, 15.0);
+            let line_end = line_start
+                + vec2(
+                    0.0,
+                    11.0 + (moves_graph_rect.height() * (1.0 - highlight.bar_height)),
+                );
+            ui.painter()
+                .line_segment([line_start, line_end], Stroke::new(1.0, Color32::WHITE));
+        }
+
+        let mut mesh = Mesh::with_texture(self.moves_graph_texture.id());
+        mesh.add_rect_with_uv(
+            moves_graph_rect,
+            Rect::from_min_max(pos2(0.0, 0.0), pos2(1.0, 1.0)),
+            Color32::WHITE,
+        );
+        ui.painter().add(Shape::mesh(mesh));
```

Comment:
Our graph is just a precalculated texture, we when we add scrolling and zooming we can just change the UV rect mapping the texture to our rect